### PR TITLE
The load balancer name must be a valid RFC 1035 label name

### PR DESCRIPTION
**Pull request by yaocw2020**
related issue: https://github.com/harvester/harvester/issues/3333

Update vendor, bump Kubernetes package to v0.23.7

### CODE REVIEW

File: pkg/cloud-controller-manager/loadBalancer.go [hunk 1]
```diff
@@ -69,15 +70,14 @@ func (l *LoadBalancerManager) GetLoadBalancerName(ctx context.Context, clusterNa
...
 	digest := crc32.ChecksumIEEE([]byte(name + string(service.UID)))
 	suffix := fmt.Sprintf("%08x", digest) // print in 8 width and pad with 0's
-	name += suffix
 
 	// The name of a Service object must be a valid [RFC 1035 label name](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names)
 	// The name contains no more than 63 characters
-	if len(name) > maxNameLength {
-		name = name[:maxNameLength]
+	if len(name) > maxNameLength-lenOfSuffix {
+		name = name[:maxNameLength-lenOfSuffix]
```
Comment (guangbochen) [inline pkg/cloud-controller-manager/loadBalancer.go:77]: can u help to add a unit test for it, also the name may still contain invalid characters like `-` after slices.

Comment (yaocw2020) [reply pkg/cloud-controller-manager/loadBalancer.go:77]: `-` is not an invalid character, but could not appear in the end.

**Review by guangbochen — APPROVED** [review]
LGTM, can we please add more details about the verification steps, thanks.

**Review by guangbochen — APPROVED** [review]

**Review by chrisho — APPROVED** [review]
LGTM and verified it, thanks
```
default     kubernetes-default-thisisthetestcasethatisbelongerthan60072c964                 dhcp
```

**Review by bk201 — APPROVED** [review]
LGTM.

**Pull request merged**